text/0000-compact-slates.md

```diff
@@ -0,0 +1,499 @@
+
+The `tx` struct in a V4 Slate is removed, and is replaced instead by the following Top-Level fields. Wallets recreate the transaction object as expected by the Grin node from these fields before posting.
+
+* `coms` is an array containing the outputs and inputs that have been added to the slate. These will appear duing the I1 or S2 phases of a transction. Each entry includes:
+   * `f`: The output features, assumed to be 'Plain' if omitted. 0 denotes Plain, 1 denotes Coinbase
+   * `c`: The output/input commitment, Base64 Encoded
+   * `p`: The output's range proof, Base64 Encoded. If this is included, the entry is assumed to be an output. If not, it is an input.
+
+When rebuilding the transaction kernel for the Node (done during the S3 or I3 phases,) the kernel is assumed to be 'Plain' unless the top-level `feat` field is non-zero. In this case, the kernel features are filled accorginly with any needed values from the `feat_args` struct.
+
+In a typical S3 phase, these fields may look something like:
+
+```
+  "coms": [
+    {
+      "c": "CA8KJv0/AmWQRgvKpTBIwxaE+OTn/TnxaADSkhrSd2yt",
+      "p": "J+/eWsEWsOljOS5norAUhV9Ygzub4Vf6dxLl71w06sA5c0QHllDeFv0n9UbCPuIL2CoBtLAfgKHAjPV7OjQ5lwSZcK+lfBD3kziiY1iUdjH6ZctAE9D/DdNfQ7a4RdurQDdKbGtihnnnOkHbllutCyV1TDLyvvqyExuwQrK2QkJulhfArQVcBE2txtbWSDjfAAtxraU7UpCMg0MoGRP4v6bm/50hntwTHc9zHegcDblADpMhTNMGh1jlRaCP00n8/Lj520YvZNd0dedynD5U97cfXFTP5C2fjVwwZKCnMZqN90CDAxNuCTA5fFaGTOv2MHMvZrqoVFriuaa+jbjpeU9ifGOWAI5B3HXMeMiSUL0a8c451ohT7zcENCr8jTW+uwA2ejcidMZsL19EaDZX9w0cirdKW5mRI/YPRuw6zBkg8SmXvSLArW0vz94pERcaE0uK9pK/qlgvplmC3/Rv92A9AkyQVeWBieyCoFaxoDL5hGMQaLLc6G3slgCzBUZHX1idNwLhfWAb5mHF97GmqcJf/joEVKuhPpK/mAICLWDUzYQS9gH4GJDlgAz1Bv5KdHY9jo/kB6IDvtxOKnTkl+vK2eK3L178vp02lufgv29fblxSFN9Kw2ujh1j8eCh8RC3RDyAI8qVbHiRDN2N4ANk0fIlc6Gq37D/VChxlTCzd+MmiBJRpqDvYFnz0r2kjuic/xATPQ+G5QoyI30ftXWgLzhGqgZ7S33rywiAP1EsgsoRwUojYvx0skfuasZd+F/Dj2o4RzX32JDcROAh6S49mcr+VumVbPx5tEcZwsQNb6d2yrzCQ6j3x5skaydyaMxhdsMtzb7BOWGWl1GrZ8AALr96R2bdP2vL4mxLtym2qZ/Hh2P5MrnivthS8C+x57TFa"
+    },
+    {
+      "c": "CA8KJv0/AmWQRgvKpTBIwxaE+OTn/TnxaADSkhrSd2yt"
+    },
+    {
+      "c": "A43jdi0/AmWQRgvKpTBIwxaE+OTn/TnxaADSkhrxer3f"
+    }
+  ],
+```
+
+#### Feature arguments
+
+Depending on the chosen Kernel Feature set, `feat_args` may be populated with
+arguments specific to the kernel. The exact arguments that will be present here
+depend on the value of 'feat'. Currently, the only supported kernel is HeightLocked
+(value 1) which has the arguments:
+
+```
+"feat_args": {
+   "lock_hgt": "2343234" // For HeightLocked kernels (1)
+}
+```
+
+### Changes from existing V3 Slate
+
+#### Top-Level Slate Struct
+
+* The `version_info` struct is removed, and is replaced with `ver`, which has the format "[version]:[block header version]"
+* `id` becomes a short-form base-64 encoding of the UUID binary
```

Comment:
Isn't the UUID hex encoded rather than base64?

Reply:
Yes, as a matter of that this is no longer a change, remove that line